corehq/apps/sms/models.py

```diff
@@ -1435,3 +1436,210 @@ def initiate_workflow(cls, domain, phone_numbers, app_id=None,
             success_numbers.append(phone_number)
 
         return (success_numbers, invalid_format_numbers, numbers_in_use)
+
+
+class SQLMobileBackend(SyncSQLToCouchMixin, models.Model):
+    SMS = 'SMS'
+    IVR = 'IVR'
+
+    TYPE_CHOICES = (
+        (SMS, ugettext_lazy('SMS')),
+        (IVR, ugettext_lazy('IVR')),
+    )
+
+    couch_id = models.CharField(max_length=126, null=True, db_index=True)
+    backend_type = models.CharField(max_length=3, choices=TYPE_CHOICES, default=SMS)
+
+    # This is an api key that the gateway uses when making inbound requests to hq.
+    # This enforces gateway security and also allows us to tie every inbound request
+    # to a specific backend.
+    inbound_api_key = models.CharField(max_length=126, unique=True)
+
+    # This tells us which type of backend this is
+    hq_api_id = models.CharField(max_length=126, null=True)
+
+    # Global backends are system owned and can be used by anyone
+    is_global = models.BooleanField(default=False)
+
+    # This is the domain that the backend belongs to, or None for
+    # global backends
+    domain = models.CharField(max_length=126, null=True, db_index=True)
+
+    # A short name for a backend instance which is referenced when
+    # setting a case contact's preferred backend
+    name = models.CharField(max_length=126)
+
+    # Simple name to display to users - e.g. "Twilio"
+    display_name = models.CharField(max_length=126, null=True)
+
+    # Optionally, a description of this backend
+    description = models.TextField(null=True)
+
+    # A JSON list of countries that this backend supports.
+    # This information is displayed in the gateway list UI.
+    # If this backend represents an international gateway,
+    # set this to: ["*"]
+    supported_countries = models.CharField(max_length=126, default='[]')
+
+    # To avoid having many tables with so few records in them, all
+    # SMS backends are stored in this same table. This field is a
+    # JSON dict which stores any additional fields that the SMS
+    # backend subclasses need.
+    extra_fields = models.TextField(default='{}')
```

Comment:
It seemed silly to have one table for each backend since each table would only have 1 record in it most of the time, so all the backends will be stored in this one model, with this `extra_fields` field containing any backend-specific data.